src/main/webapp/user-private-comments.jsp

```diff
@@ -23,10 +24,10 @@
         <%}%>
           <%=task.getName()%>
         </button>
-        <form class="inline" id="add-private-comment-post-form" action="/user-profile" method="POST">
-          <input type="hidden" id="add-private-comment-task-input" name="taskID" value="<%=task.getTaskID()%>">
+        <form action="/user-profile" method="POST">
```

Comment:
You have multiples of these on the page I believe (one per task), but all of them have the inputs/textarea have the same name "taskID" and "message".  I'm not sure how HTTP would handle this, whether it'll send a list of if it'll send just the first one.  Try to look at the Chrome Dev Console so see what the request looks like.


Also since you're submitting all of them into the same user-profile servlet, that means it'll try to resave all messages into datastore, which would then be inefficient if there are comments that were not updated.

Try to see if you can build it so each comment would submit (and save in datastore) individually.